kimchi/src/circuits/expr.rs

```diff
@@ -1921,14 +2105,23 @@ impl<F: Neg<Output = F> + Clone + One + Zero + PartialEq> Expr<F> {
                 let x = x.monomials(ev);
                 mul_monomials(&x, &x)
             }
-            EnabledIf(feature, x) => {
-                let mut monomials = x.monomials(ev);
-                for expr in monomials.values_mut() {
-                    let mut unflagged = Expr::zero();
-                    std::mem::swap(expr, &mut unflagged);
-                    *expr = Expr::EnabledIf(*feature, Box::new(unflagged))
+            IfFeature(feature, e1, e2) => {
+                let mut res = HashMap::new();
+                let e1_monomials = e1.monomials(ev);
+                let mut e2_monomials = e2.monomials(ev);
+                for (m, c) in e1_monomials.into_iter() {
+                    let else_branch = match e2_monomials.remove(&m) {
+                        None => Expr::zero(),
+                        Some(c) => c,
+                    };
+                    let expr = Expr::IfFeature(*feature, Box::new(c), Box::new(else_branch));
+                    res.insert(m, expr);
                 }
-                monomials
+                for (m, c) in e2_monomials.into_iter() {
+                    let expr = Expr::IfFeature(*feature, Box::new(Expr::zero()), Box::new(c));
```

Comment:
Why is this always zero in the first branch, whereas in the other it can be either zero or `c`?